x-pack/plugins/uptime/public/components/monitor_management/show_sync_errors.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { kibanaService } from '../../state/kibana_service';
+import { toMountPoint } from '../../../../../../src/plugins/kibana_react/public';
+import { ServiceLocationErrors, ServiceLocations } from '../../../common/runtime_types';
+
+export const showSyncErrors = (errors: ServiceLocationErrors, locations: ServiceLocations) => {
+  Object.values(errors).forEach((location) => {
+    const { status: responseStatus, reason } = location.error || {};
+    kibanaService.toasts.addWarning({
+      title: i18n.translate('xpack.uptime.monitorManagement.service.error.title', {
+        defaultMessage: `Unable to sync monitor config`,
+      }),
+      text: toMountPoint(
+        <>
+          <p>
+            {i18n.translate('xpack.uptime.monitorManagement.service.error.message', {
+              defaultMessage: `Your monitor was saved, but there was a problem syncing the configuration for {location}. We will automatically try again later. If this problem continues, your monitors will stop running in {location}. Please contact Support for assistance.`,
```

Comment:
IMO it would eventually be better to have a single toast mentioning all locations where it's not working rather than one toast per location, but I won't block because of that.

Reply:
Yes, I see where you're coming from. In reality, this isn't so much related to this feature in particular. 

> LGTM, only small nits.
> 
> Also tested E2E and was able to see the warning after changing the auth so that I can't auth with the service anymore, as shown in the screenshot below.
> 
> ![Screenshot 2022-03-29 at 21 43 55](https://user-images.githubusercontent.com/6868147/160703905-3e8fd335-596e-4235-b88b-7575b9025913.png)
> 
> ⚠️ My only concern is that in my test the warning says `undefined` in the box on the top. Considering FF is today I think we can get this in and fix the location name bug post FF, that's why I'm marking this as approved. Otherwise, I'd "request changes".
> 
> I'll defer to you on whether you think this should be blocking the merge.
> 
> > Please notice I tested against the remote dev service using `ngrok` as instructed in the PR.
> 
> PS: Didn't know about the configurable interval, that's great!

The location undefined issues happens when locations are not loaded beforehand. I will block this screen from rendering until locations is loaded. I thought that was implemented correctly as preventing the screen from rendering until locations was loaded should have been addressed in a separate PR that was merged a few hours ago. However, there may have been an issue with merge conflicts after merging upstream.